client/homebrew/pages/userPage/userPage.jsx

```diff
@@ -14,69 +14,50 @@ const HelpNavItem = require('../../navbar/help.navitem.jsx');
 const ErrorNavItem = require('../../navbar/error-navitem.jsx');
 const VaultNavitem = require('../../navbar/vault.navitem.jsx');
 
-const UserPage = createClass({
-	displayName     : 'UserPage',
-	getDefaultProps : function() {
-		return {
-			username : '',
-			brews    : [],
-			query    : '',
-			error    : null
-		};
-	},
-	getInitialState : function() {
-		const usernameWithS = this.props.username + (this.props.username.endsWith('s') ? `’` : `’s`);
-
-		const brews = _.groupBy(this.props.brews, (brew)=>{
-			return (brew.published ? 'published' : 'private');
-		});
-
-		const brewCollection = [
-			{
-				title : `${usernameWithS} published brews`,
-				class : 'published',
-				brews : brews.published
-			}
-		];
-		if(this.props.username == global.account?.username){
-			brewCollection.push(
-				{
-					title : `${usernameWithS} unpublished brews`,
-					class : 'unpublished',
-					brews : brews.private
-				}
-			);
-		}
-
-		return {
-			brewCollection : brewCollection
-		};
-	},
-	errorReported : function(error) {
-		this.setState({
-			error
-		});
-	},
-
-	navItems : function() {
-		return <Navbar>
-			<Nav.section>
-				{this.state.error ?
-					<ErrorNavItem error={this.state.error} parent={this}></ErrorNavItem> :
-					null
-				}
-				<NewBrew />
-				<HelpNavItem />
-				<VaultNavitem/>
-				<RecentNavItem />
-				<Account />
-			</Nav.section>
-		</Navbar>;
-	},
-
-	render : function(){
-		return <ListPage brewCollection={this.state.brewCollection} navItems={this.navItems()} query={this.props.query} reportError={this.errorReported}></ListPage>;
-	}
-});
+const UserPage = (props) => {
+    const { username = '', brews = [], query = '', error = null } = props;
+    const usernameWithS = username + (username.endsWith('s') ? `’` : `’s`);
+
+    const groupedBrews = _.groupBy(brews, (brew) => {
+        return brew.published ? 'published' : 'private';
+    });
+
+	const brewCollection = [
+		{
+			title: `${usernameWithS} published brews`,
+			class: 'published',
+			brews: groupedBrews.published || []
+		},
+		...(username === global.account?.username ? [{
+			title: `${usernameWithS} unpublished brews`,
+			class: 'unpublished',
+			brews: groupedBrews.private || []
+		}] : [])
+	];	
+
+    const [currentError, setCurrentError] = useState(error);
+
+    const errorReported = (error) => {
```

Comment:
Can this be renamed to `handleReportError`?  My gut feeling is that this would be easier to understand as a function handling a child component prop, following the same convention elsewhere.  As-is, it seems like a regular "local helper" function like `usernameWithS`.

In fact, we could likely get rid of the function entirely by just setting up `ListPage` like this:

```
<ListPage brewCollection={brewCollection}  navItems={navItems()} query={query} reportError={()=>setCurrentError(error)} />
```

*note, you can also throw the last `/>` on line 59 on the same line as the rest of it since it's not doing much on it's own, and reduce your lines of code by 1*